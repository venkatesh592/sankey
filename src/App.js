import React from 'react';
import { Sankey } from './features/sankey/Sankey';
import './App.css';
import { Header } from './features/header/Header';

function App() {
  return (
    <div className="App">
      <Header title='Centime' ></Header>
      <Sankey />
    </div>
  );
}

export default App;
