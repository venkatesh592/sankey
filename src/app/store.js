import { configureStore } from '@reduxjs/toolkit';
import sankeyReducer from '../features/sankey/sankeySlice';

export default configureStore({
  reducer: {
    options: sankeyReducer,
  },
});
