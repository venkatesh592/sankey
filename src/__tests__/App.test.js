import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../app/store';
import App from '../App';

let container = null
beforeEach(() => {
  container = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
})

test('renders With out fail', () => {
  expect(container.getByText('Centime')).toBeInTheDocument();
});

test('snapshot', () => {
  expect(container.asFragment()).toMatchSnapshot();
});
