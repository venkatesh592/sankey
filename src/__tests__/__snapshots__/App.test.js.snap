// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="header"
    >
      <a
        href="#default"
      >
        Centime
      </a>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline select"
        data-testid="slelect-lang"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="demo-simple-select"
          role="button"
          tabindex="0"
        >
          English
        </div>
        <input
          type="hidden"
          value="en"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="sankey"
    >
      <div
        class="usage"
      >
        Usage: (Feed the Data)
        <br />
        1) Create Data point (should be unique)
        <br />
        2) Create Link between them
        <br />
        Example: Salary &gt; Home [3000] &gt; [Electricity - 1000, Grocery - 2000]
        <br />
        Steps:
        <br />
        1) Create four data points with names like [Salary, Home, Electricity, Grocery]
        <br />
        2) Create three links between them
        <br />
        a) source: Salary, target: Home, Expenditure: 3000 b) source: Home, target: Electricity, Expenditure: 1000 c) source: Home, target: Grocery, Expenditure: 2000
        <br />
        You can see graph below based on inputs. (you can also do edit and delete options as well)
        <br />
        Note: Description only availalebe in English
      </div>
      <div
        class="container"
      >
        <div
          class="dataPoint"
        >
          <span>
            Create Entry
          </span>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline input"
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Usage Name Placeholder"
              type="text"
              value=""
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Create
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="dataLink"
        >
          <span>
            Link Expenditure
          </span>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline input"
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Source Placeholder"
              type="text"
              value=""
            />
          </div>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline input"
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Target Placeholder"
              type="text"
              value=""
            />
          </div>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline input"
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Expenditure"
              type="text"
              value=""
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Create
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="container"
      />
    </div>
  </div>
</DocumentFragment>
`;
